qiskit/pulse/instructions/shift_phase.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The shift phase instruction updates the modulation phase of pulses played on a channel."""
+
+import warnings
+
+from typing import List, Optional, Union
+
+from qiskit.pulse.channels import PulseChannel
+from qiskit.pulse.exceptions import PulseError
+from .instruction import Instruction
+
+
+class ShiftPhase(Instruction):
+    """The shift phase instruction updates the modulation phase of proceeding pulses played on the
+    same :py:class:`~qiskit.pulse.channels.Channel`. It is a relative increase in phase determined
+    by the ``phase`` operand.
+
+    The qubit phase is tracked in software, enabling instantaneous, nearly error-free Z-rotations
+    by using a ShiftPhase to update the frame tracking the qubit state.
+    """
+
+    def __init__(self, phase: float,
+                 channel: Optional[PulseChannel] = None,
+                 name: Optional[str] = None):
+        """Instantiate a shift phase instruction, increasing the output signal phase on ``channel``
+        by ``phase`` [radians].
+
+        Args:
+            phase: The rotation angle in radians.
+            channel: The channel this instruction operates on.
+            name: Display name for this instruction.
+        """
+        if channel is None:
+            warnings.warn("Usage of ShiftPhase without specifying a channel is deprecated. For "
+                          "example, ShiftPhase(3.14)(DriveChannel(0)) should be replaced by "
+                          "ShiftPhase(3.14, DriveChannel(0)).", DeprecationWarning)
+        self._phase = phase
+        self._channel = channel
+        super().__init__(0, channel, name=name or 'fc{:x}'.format(hash((phase, channel)))[:10])
```

Comment:
Why do you change the default naming behavior? The previous naming behavior worked well?

Reply:
The change is the way the arbitrary number to be appended to 'fc' is generated. The format is still the same, but this method for generation reduces the SNR

Reply:
I'm not sure how this reduces the SNR? Doesn't this introduce a second way of doing automated naming, increasing the complexity.

Reply:
The reduction is from 1:21 to 1:1 for the number of lines used to generate a number that has no significance.

Reply:
The number does have significance as it is a consistent placeholder strategy for every instruction. I would recommend changing every instruction, or only this instruction.